lib/internal/histogram.js

```diff
@@ -101,39 +194,98 @@ class Histogram extends JSTransferable {
     return this[kHandle]?.percentile(percentile);
   }
 
+  /**
+   * @param {number} percentile
+   * @returns {bigint}
+   */
+  percentileBigInt(percentile) {
+    if (!isHistogram(this))
+      throw new ERR_INVALID_THIS('Histogram');
+    validateNumber(percentile, 'percentile');
+
+    if (NumberIsNaN(percentile) || percentile <= 0 || percentile > 100)
+      throw new ERR_INVALID_ARG_VALUE.RangeError('percentile', percentile);
+
+    return this[kHandle]?.percentileBigInt(percentile);
+  }
+
+  /**
+   * @readonly
+   * @type {Map<number,number>}
+   */
   get percentiles() {
+    if (!isHistogram(this))
+      throw new ERR_INVALID_THIS('Histogram');
     this[kMap].clear();
     this[kHandle]?.percentiles(this[kMap]);
     return this[kMap];
   }
 
-  reset() {
-    this[kHandle]?.reset();
+  /**
+   * @readonly
+   * @type {Map<number,bigint>}
+   */
+  get percentilesBigInt() {
+    if (!isHistogram(this))
+      throw new ERR_INVALID_THIS('Histogram');
+    this[kMap].clear();
+    this[kHandle]?.percentilesBigInt(this[kMap]);
+    return this[kMap];
   }
 
-  [kDestroy]() {
-    this[kHandle] = undefined;
+  /**
+   * @returns {void}
+   */
+  reset() {
+    if (!isHistogram(this))
+      throw new ERR_INVALID_THIS('Histogram');
+    this[kHandle]?.reset();
   }
 
   [kClone]() {
     const handle = this[kHandle];
     return {
       data: { handle },
-      deserializeInfo: 'internal/histogram:InternalHistogram'
+      deserializeInfo: 'internal/histogram:internalHistogram'
     };
   }
 
   [kDeserialize]({ handle }) {
     this[kHandle] = handle;
   }
+
+  toJSON() {
+    return {
+      count: this.count,
+      min: this.min,
+      max: this.max,
+      mean: this.mean,
+      exceeds: this.exceeds,
+      stddev: this.stddev,
+      percentiles:
+        ArrayPrototypeReduce(
+          ArrayFrom(MapPrototypeEntries(this.percentiles)),
+          (obj, entry) => {
+            obj[entry[0]] = entry[1];
+            return obj;
+          }, {}
+        )
```

Comment:
We can use `ObjectFromEntries()` here instead of `ArrayPrototypeReduce()` here in this case for more readability.

Reply:
Oh right! I forgot all about `Object.fromEntries()`